api/src/main/java/org/openmrs/module/kenyaemr/reporting/data/converter/definition/evaluator/art/NCDSDataEvaluator.java

```diff
@@ -33,39 +33,40 @@ public class NCDSDataEvaluator implements PersonDataEvaluator {
     public EvaluatedPersonData evaluate(PersonDataDefinition definition, EvaluationContext context) throws EvaluationException {
         EvaluatedPersonData c = new EvaluatedPersonData(definition, context);
 
-        String qry = "\n" +
-                "select ci.patient_id,group_concat( case ci.chronic_illness\n" +
-                "                         when 149019 then 'Alzheimers Disease and other Dementias'\n" +
-                "                         when 148432 then 'Arthritis'\n" +
-                "                         when 153754 then 'Asthma'\n" +
-                "                         when 159351 then 'Cancer'\n" +
-                "                         when 119270 then 'Cardiovascular diseases'\n" +
-                "                         when 120637 then 'Chronic Hepatitis'\n" +
-                "                         when 145438 then 'Chronic Kidney Disease'\n" +
-                "                         when 1295 then 'Chronic Obstructive Pulmonary Disease(COPD)'\n" +
-                "                         when 120576 then 'Chronic Renal Failure'\n" +
-                "                         when 119692 then 'Cystic Fibrosis'\n" +
-                "                         when 120291 then 'Deafness and Hearing impairment'\n" +
-                "                         when 119481 then 'Diabetes'\n" +
-                "                         when 118631 then 'Endometriosis'\n" +
-                "                         when 117855 then 'Epilepsy'\n" +
-                "                         when 117789 then 'Glaucoma'\n" +
-                "                         when 139071 then 'Heart Disease'\n" +
-                "                         when 115728 then 'Hyperlipidaemia'\n" +
-                "                         when 117399 then 'Hypertension'\n" +
-                "                         when 117321 then 'Hypothyroidism'\n" +
-                "                         when 151342 then 'Mental illness'\n" +
-                "                         when 133687 then 'Multiple Sclerosis'\n" +
-                "                         when 115115 then 'Obesity'\n" +
-                "                         when 114662 then 'Osteoporosis'\n" +
-                "                         when 117703 then 'Sickle Cell Anaemia'\n" +
-                "                         when 118976 then 'Thyroid disease'\n" +
-                "                         when 141623 then 'Dyslipidemia'\n" +
-                "              end SEPARATOR\n" +
-                "              ',') as ChronicIllness\n" +
-                "              from kenyaemr_etl.etl_allergy_chronic_illness ci\n" +
-                "              where ci.visit_date <= date(:endDate)\n" +
-                "              group by ci.patient_id;";
+        String qry = "SELECT ci.patient_id,\n" +
```

Comment:
You need to handle a case where there are more than on NCD recorded in that visit

Reply:
Resolved. Thanks